polkadot/node/core/dispute-coordinator/src/scraping/mod.rs

```diff
@@ -113,10 +113,13 @@ impl Inclusions {
 	/// The candidates at the block height are NOT removed.
 	pub fn remove_up_to_height(&mut self, height: &BlockNumber) {
 		let not_stale = self.candidates_by_block_number.split_off(&height);
-		let stale = std::mem::take(&mut self.candidates_by_block_number);
+		let stale: HashSet<_> = std::mem::take(&mut self.candidates_by_block_number)
+			.into_values()
+			.flatten()
+			.collect();
```

Comment:
Very good catch! Given that this should indeed be pretty rare, I would avoid the inefficiency of collecting into a `HashMap` and instead remove the debug assert and the debug log. 

Really not a strong concern though, also with asynchronous backing this can happen more often.